installer/block_device.go

```diff
@@ -433,3 +433,13 @@ func (fw *FlushingWriter) Sync() error {
 	fw.unflushedBytesWritten = 0
 	return err
 }
+
+func (fw *FlushingWriter) Close() error {
+	sync_err := fw.Sync()
+	close_err := fw.BlockDevicer.Close()
```

Comment:
Q: Isn't this pretty much the same as defering the blockdevice closing?

Reply:
Pretty much, but it's trickier to get the error returned correctly with `defer`, especially with the two possibilities.